File: frontend/lib/models/datos_ambientales.dart
import 'package:json_annotation/json_annotation.dart';

part 'datos_ambientales.g.dart';

@JsonSerializable()
class DatosAmbientales {
  final int? idRegistro;
  final DateTime? timestamp;
  final String? fuenteDatos;
  final double? radiacionGlobalHoriz_Wh_m2;
  final double? temperaturaAmbiente_C;
  final double? velocidadViento_m_s;
  final int? idSimulacion;

  DatosAmbientales({
    this.idRegistro,
    this.timestamp,
    this.fuenteDatos,
    this.radiacionGlobalHoriz_Wh_m2,
    this.temperaturaAmbiente_C,
    this.velocidadViento_m_s,
    this.idSimulacion,
  });

  factory DatosAmbientales.fromJson(Map<String, dynamic> json) => _$DatosAmbientalesFromJson(json);
  Map<String, dynamic> toJson() => _$DatosAmbientalesToJson(this);
}

File: backend/app/domain/entities/datos_ambientales.py
from dataclasses import dataclass
from typing import Optional
from datetime import datetime

@dataclass
class DatosAmbientalesEntity:
    idRegistro: int = None
    timestamp: datetime = None
    fuenteDatos: str = None
    radiacionGlobalHoriz_Wh_m2: float = None
    temperaturaAmbiente_C: float = None
    velocidadViento_m_s: float = None
    idSimulacion: int = None